pkg/didcomm/dispatcher/outbound.go

```diff
@@ -15,29 +15,55 @@ import (
 	commontransport "github.com/hyperledger/aries-framework-go/pkg/didcomm/common/transport"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
 )
 
 // provider interface for outbound ctx
 type provider interface {
 	Packager() commontransport.Packager
 	OutboundTransports() []transport.OutboundTransport
 	TransportReturnRoute() string
+	VDRIRegistry() vdri.Registry
 }
 
 // OutboundDispatcher dispatch msgs to destination
 type OutboundDispatcher struct {
 	outboundTransports   []transport.OutboundTransport
 	packager             commontransport.Packager
 	transportReturnRoute string
+	vdRegistry           vdri.Registry
 }
 
+const (
+	ed25519KeyType     = "Ed25519VerificationKey2018"
+	didCommServiceType = "did-communication"
+)
+
 // NewOutbound return new dispatcher outbound instance
 func NewOutbound(prov provider) *OutboundDispatcher {
 	return &OutboundDispatcher{
 		outboundTransports:   prov.OutboundTransports(),
 		packager:             prov.Packager(),
 		transportReturnRoute: prov.TransportReturnRoute(),
+		vdRegistry:           prov.VDRIRegistry(),
+	}
+}
+
+// SendToDID sends a message from myDID to the agent who owns theirDID
+func (o *OutboundDispatcher) SendToDID(msg interface{}, myDID, theirDID string) error {
```

Comment:
This function also can be optimized. We can get myDID from `DIDConnectionStore` by theirDID. It would be easy to use.

Reply:
Public DIDs, which we can use multiple peer DIDs to communicate with, means we can't actually do this mapping in all cases. I've added support for providing "myDID" to the inbound handler in a service.